### PR TITLE
Fix replication issue with index ID/name conflicts

### PR DESCRIPTION
**Pull request by danielhlarkin**

### CODE REVIEW

**Comment by danielhlarkin** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/3511/

File: tests/js/server/replication/replication-sync.js [hunk 1]
```diff
@@ -215,6 +217,153 @@ function BaseTestConfig () {
...
+        },
+        function (state) {
+          //  already create the collection and index on the slave
+          const c = db._create(cn);
+          c.ensureIndex(state.indexDef);
+        },
+        function (state) {
+          const c = db._collection(cn);
+          const i = c.index(state.indexDef.id);
+          assertEqual(state.indexProps.id, i.id);
```
Comment (jsteemann) [inline tests/js/server/replication/replication-sync.js]: should we assert here that index names be identical too?
(in this test and the following)

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3516/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3524/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3527/

**Pull request merged**